### PR TITLE
Add checking of dependencies consistency

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: 2.0/gradle/vividus-test.gradle [hunk 1]
```diff
@@ -158,6 +158,63 @@ tasks.withType(StoriesExec).configureEach {
...
+        if (vividusLib == null) {
+            throw new GradleException('The mandatory \'vividus\' dependency is missing in project dependencies'
+                + ' configuration.')
+        }
+
+        def vividusBomLib = dependencies.find { nameMatcher.call(it, 'vividus-bom') }
+        if (vividusBomLib != null) {
+            def vividusBomVersion = vividusBomLib.version
+            if (vividusBomVersion == null) {
+                throw new GradleException('Please specify version for \'vividus-bom\' dependency in project'
```
Comment (valfirst) [inline 2.0/gradle/vividus-test.gradle:188]: won't it fail earlier?

File: 2.0/gradle/vividus-test.gradle [hunk 2]
```diff
@@ -158,6 +158,63 @@ tasks.withType(StoriesExec).configureEach {
...
+        }
+
+        ext.vividusVersion = vividusLib.version
+   
+        def dependenciesWithInvalidVersions = dependencies.findAll {
+            groupMatcher.call(it) && !vividusVersion.equals(it.version)
+        }.collect { it.name }
+
+        if (!dependenciesWithInvalidVersions.isEmpty()) {
+            throw new GradleException("The follwing dependencies are expected to have ${vividusVersion} version: "
```
Comment (valfirst) [inline 2.0/gradle/vividus-test.gradle:212]:
```suggestion
            throw new GradleException("The following dependencies must be of ${vividusVersion} version: "
```

**Review by valfirst — CHANGES_REQUESTED** [review]

File: 2.0/gradle/vividus-test.gradle (hunk 1 above)
Comment (uarlouski) [reply 2.0/gradle/vividus-test.gradle:188]: suprisingly its not, only for impementation dependencies

Comment (uarlouski) [reply 2.0/gradle/vividus-test.gradle:188]: discussed

**Review by valfirst — APPROVED** [review]

**Pull request merged**